src/core/Akka.Remote.TestKit/DataTypes.cs

```diff
@@ -277,35 +270,37 @@ public override int GetHashCode()
 
         public Hello(string name, Address address)
         {
-            _name = name;
-            _address = address;
+            Name = name;
+            Address = address;
         }
 
-        public string Name
-        {
-            get { return _name; }
-        }
+        public string Name { get; }
+
+        public Address Address { get; }
 
-        public Address Address
+        public override string ToString()
         {
-            get { return _address; }
+            return $"Hello(Name: {Name}, Address: {Address})";
         }
     }
 
-    sealed class EnterBarrier : IServerOp, INetworkOp
+    internal sealed class EnterBarrier : IServerOp, INetworkOp
     {
-        readonly string _name;
-        readonly TimeSpan? _timeout;
+        public EnterBarrier(string name, TimeSpan? timeout, RoleName role)
```

Comment:
Key diagnostic improvement: add the `RoleName` of the node trying to cross the barrier to these messages.